app/src/main/java/ml/docilealligator/infinityforreddit/comment/ParseComment.java

```diff
@@ -52,17 +55,75 @@ public static void parseComment(Executor executor, Handler handler, String respo
     }
 
     static void parseMoreComment(Executor executor, Handler handler, String response,
-                                 ArrayList<Comment> commentData, int depth, boolean expandChildren,
+                                 ArrayList<Comment> commentData, boolean expandChildren,
                                  ParseCommentListener parseCommentListener) {
         executor.execute(() -> {
             try {
-                JSONArray childrenArray = new JSONObject(response).getJSONObject(JSONUtils.DATA_KEY).getJSONArray(JSONUtils.CHILDREN_KEY);
+                JSONArray childrenArray = new JSONObject(response).getJSONObject(JSONUtils.JSON_KEY)
+                        .getJSONObject(JSONUtils.DATA_KEY).getJSONArray(JSONUtils.THINGS_KEY);
 
                 ArrayList<Comment> newComments = new ArrayList<>();
                 ArrayList<Comment> expandedNewComments = new ArrayList<>();
-                ArrayList<String> moreChildrenFullnames = new ArrayList<>();
+                ArrayList<String> moreChildrenIds = new ArrayList<>();
+
+                // api response is a flat list of comments tree
+                // process it in order and rebuild the tree
+                for (int i = 0; i < childrenArray.length(); i++) {
+                    JSONObject child = childrenArray.getJSONObject(i);
+                    JSONObject childData = child.getJSONObject(JSONUtils.DATA_KEY);
+                    if (child.getString(JSONUtils.KIND_KEY).equals(JSONUtils.KIND_VALUE_MORE)) {
+                        String parentFullName = childData.getString(JSONUtils.PARENT_ID_KEY);
+                        JSONArray childrenIds = childData.getJSONArray(JSONUtils.CHILDREN_KEY);
+
+                        if (childrenIds.length() != 0) {
+                            ArrayList<String> localMoreChildrenIds = new ArrayList<>(childrenIds.length());
+                            for (int j = 0; j < childrenIds.length(); j++) {
+                                localMoreChildrenIds.add(childrenIds.getString(j));
+                            }
+
+                            Comment parentComment = findCommentByFullName(newComments, parentFullName);
+                            if (parentComment != null) {
+                                parentComment.setHasReply(true);
```

Comment:
Setting `hasReply` manually feels weird because if I understood it correctly, it is always equal to `!children.isEmpty() || !moreChildren.isEmpty()`